### PR TITLE
1235 angr on android

### PR DESCRIPTION
**Pull request by commjoen**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [ ] Your contribution is written in the 2nd person (e.g. you)
- [ ] Your contribution is written in an active present form for as much as possible.
- [ ] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [ ] Your contribution has proper formatted markdown and/or code
- [ ] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ ] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #1235 .

### CODE REVIEW

**Comment by OMTGreleaser** [comment]
Results for commit da5979375ad86fe2f46ad0cca3d5b11484b05f9b: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit edd00b319b54378861c699846751ccfa50584169: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        5 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 28bd1416070c137353031eed2358399ee8f9589e: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 16ae1c9ab41d9ec60bdf8b76c4ee86a58790b885: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05b-Basic-Security_Testing.md [hunk 1]
```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
...
 ```
 
-Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else.
+Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else. Alternatively, there is always a docker container which you can use.
 
 Comprehensive documentation, including an installation guide, tutorials, and usage examples is available on [Gitbooks page of angr](https://docs.angr.io/ "angr"). A complete [API reference](https://angr.io/api-doc/ "angr API") is also available.
 
+You can use angr with a pythorn REPL - such as iPython - or script your approaches. Although angr has a bit of a steep learning curve, we do recommend using it when you want to brute force your way to a given state of an executable. See [Binary Analysis Frameworks](###Binary Analysis Frameworks) as a great example of how this can work.
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]: Fixes typo, section name (actually is Binary Analysis now, but angr is in Symbolic Execution), also referencing with several #'s doesn't work. only 1 is required

```suggestion
You can use angr from a Python REPL - such as iPython - or script your approaches. Although angr has a bit of a steep learning curve, we do recommend using it when you want to brute force your way to a given state of an executable. See section [Symbolic Execution](#Symbolic-Execution) as a great example on how this can work.
```

File: Document/0x05b-Basic-Security_Testing.md [hunk 2]
```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
...
 
 Since version 8 Angr is based on Python 3, and it's available from PyPI. With pip, it's easy to install on \*nix operating systems and Mac OS:
 
 ```shell
-$ pip install angr
+$ mkvirtualenv --python=$(which python3) angr && python -m pip install angr
 ```
 
-Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else.
+Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else. Alternatively, there is always a docker container which you can use.
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]: As said in the prev. comment:

```suggestion
> Some of angr's dependencies contain forked versions of the Python modules Z3 and PyVEX, which would overwrite the original versions. If you're using those modules for anything else, you should create a dedicated virtual environment with Virtualenv. Alternatively, you can always use the provided docker container. See the [installation guide](https://docs.angr.io/introductory-errata/install "angr Installation Guide") for more details.
```

**Comment by OMTGreleaser** [comment]
Results for commit 9848497c575a60d5b0dd0859f4d585f32a1dc5f5: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05b-Basic-Security_Testing.md [hunk 3]
```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
...
 
-Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with native Android binaries.
+Angr allows for disassempbly, program instrumentation, symbolic execution, control-flow analysis, data-dependency analysis, decompilation and more, given a large set of plugins.
 
 Since version 8 Angr is based on Python 3, and it's available from PyPI. With pip, it's easy to install on \*nix operating systems and Mac OS:
 
 ```shell
-$ pip install angr
+$ mkvirtualenv --python=$(which python3) angr && python -m pip install angr
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]: I'd put the virtualenv stuff in a note and just leave "pip install angr" here.

```suggestion
$ pip install angr
```

File: Document/0x05b-Basic-Security_Testing.md [hunk 4]
```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
...
 ```
 
-Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else.
+Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else. Alternatively, there is always a docker container which you can use.
 
 Comprehensive documentation, including an installation guide, tutorials, and usage examples is available on [Gitbooks page of angr](https://docs.angr.io/ "angr"). A complete [API reference](https://angr.io/api-doc/ "angr API") is also available.
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]:
```suggestion
Comprehensive documentation, including an installation guide, tutorials, and usage examples are available on [Gitbooks page of angr](https://docs.angr.io/ "angr"). A complete [API reference](https://angr.io/api-doc/ "angr API") is also available.
```

File: Document/0x05b-Basic-Security_Testing.md [hunk 5]
```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
 
 ##### Angr
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1235>
+Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. In other words: given a binary and a requested state, Angr will try to get to that to the requested state in the binary, using formal methods to find a path, as well as brute forcing. This can make reverse-engineerig "I just want a solution style" - way easier than with for instance IDA Pro, Radare2, or Ghidra. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with _native_ Android binaries.
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]: - This can make reverse-engineerig "I just want a solution style" - way easier than with for instance IDA Pro, Radare2, or Ghidra. -> two things: 1) I would rephrase the first sentence but don't know why now. 2) why is it easier?

- Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with _native_ Android binaries. -> do you mean Android native code? Maybe: "Android native code such as native libraries"?

File: Document/0x05b-Basic-Security_Testing.md [hunk 6]
```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
...
 
-Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with native Android binaries.
+Angr allows for disassempbly, program instrumentation, symbolic execution, control-flow analysis, data-dependency analysis, decompilation and more, given a large set of plugins.
 
 Since version 8 Angr is based on Python 3, and it's available from PyPI. With pip, it's easy to install on \*nix operating systems and Mac OS:
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]:
```suggestion
Since version 8, Angr is based on Python 3, and can be installed with pip on \*nix operating systems, macOS and Windows:
```

File: Document/0x05b-Basic-Security_Testing.md [hunk 7]
```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
 
 ##### Angr
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1235>
+Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. In other words: given a binary and a requested state, Angr will try to get to that to the requested state in the binary, using formal methods to find a path, as well as brute forcing. This can make reverse-engineerig "I just want a solution style" - way easier than with for instance IDA Pro, Radare2, or Ghidra. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with _native_ Android binaries.
 
-Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with native Android binaries.
+Angr allows for disassempbly, program instrumentation, symbolic execution, control-flow analysis, data-dependency analysis, decompilation and more, given a large set of plugins.
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]:
```suggestion
Angr allows for disassembly, program instrumentation, symbolic execution, control-flow analysis, data-dependency analysis, decompilation and more, given a large set of plugins.
```

**Comment by OMTGreleaser** [comment]
Results for commit a8bffe02e9cf33a7926f8748ebcf57f90218eefd: Broken link result: [!] Error(s) found by the Links validator: 6 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05b-Basic-Security_Testing.md (hunk 5 above)
Comment (commjoen) [reply Document/0x05b-Basic-Security_Testing.md]: 1.: Is `This can make reverse-engineerig "I just want a solution style" - way easier than with for instance IDA Pro, Radare2, or Ghidra.`  --> `Using angr to get to the requested state is way faster than taking manual steps for debugging and searching the path towards the required state.` better?
2.: yes, i meant Android native code such as native libraries. Will fix that

**Comment by OMTGreleaser** [comment]
Results for commit 18fb7f465378f98a7b429fc668bf7b941f52161a: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05b-Basic-Security_Testing.md (hunk 5 above)
Comment (cpholguera) [reply Document/0x05b-Basic-Security_Testing.md]: Yes, I like that!

**Comment by OMTGreleaser** [comment]
Results for commit a07a5f8c832ea588eb1c205b26b620fe45fdd9f4: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05b-Basic-Security_Testing.md [hunk 8]
```diff
@@ -247,19 +248,21 @@ You'll come across different use cases on how you can use adb commands when test
 
 ##### Angr
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1235>
+Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. In other words: given a binary and a requested state, Angr will try to get to that to the requested state in the binary, using formal methods to find a path, as well as brute forcing. Using angr to get to the requested state is way faster than taking manual steps for debugging and searching the path towards the required state. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with native code, such as native Android binaries.
```
Comment (cpholguera) [inline Document/0x05b-Basic-Security_Testing.md]:
```suggestion
Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. In other words: given a binary and a requested state, angr will try to get to the requested state in the binary, using formal methods (a technique used for static code analysis) to find a path, as well as brute forcing. Using angr to get to the requested state is way faster than taking manual steps for debugging and searching the path towards the required state. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with native code, such as native Android binaries.
```

**Comment by OMTGreleaser** [comment]
Results for commit 24acabf17f8fdf0fa36e6b06b381ae34ca06341d: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 85595bc7c9ef184aecaa4dd4a3e3620a17c64bfc: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit fa9f9e3ff7ef05a722191ce1aad0ef8a0bbe6d7a: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.
